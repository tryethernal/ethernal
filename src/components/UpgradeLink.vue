<template>
    <a @click.stop="goToBilling()"><slot>Upgrade</slot></a>
</template>
<script>
export default {
    name: 'UpgradeLink',
    props: ['emit'],
    methods: {
        goToBilling: function() {
            if (this.$emit)
                this.$emit('goToBilling');
            else
                this.$router.push({ path: '/settings', query: { tab: 'billing' }});
        }
    }
}
</script>
<style scoped>
a {
    text-decoration: underline;
}
</style>
