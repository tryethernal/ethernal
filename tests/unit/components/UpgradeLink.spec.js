import MockHelper from '../MockHelper';

import UpgradeLink from '@/components/UpgradeLink.vue';

describe('UpgradeLink.vue', () => {
    let helper;

    beforeEach(() => {
        helper = new MockHelper();
    });

    it('Should display a link to the billing page', (done) => {
        const wrapper = helper.mountFn(UpgradeLink);
        expect(wrapper.html()).toMatchSnapshot();
        done();
    });
});
