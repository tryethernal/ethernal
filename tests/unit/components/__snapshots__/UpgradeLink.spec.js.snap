// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpgradeLink.vue Should display a link to the billing page 1`] = `<a>Upgrade</a>`;
